tests/visualization/test_graphviz.py

```diff
@@ -150,3 +150,47 @@ def test_filename(self):
         self.assertTrue(os.path.isfile("test_graphviz_filename.svg"))
         if not SAVE_IMAGES:
             self.addCleanup(os.remove, "test_graphviz_filename.svg")
+    
+    def test_escape_sequences(self):
+        # Create a simple graph
+        graph = rustworkx.generators.path_graph(2)
+        
+        # List of escape sequences to test
+        escapes = ["\\n", "\\t", "\\a"]
+        
+        # Define node attributes including randomly inserted escape sequences
+        def node_attr(node):
+            # Create base strings
+            base_label = "label"
+            base_tooltip = "tooltip"
+            
+            # Insert escape sequences at random positions
+            insert_at_label = random.randint(0, len(base_label))
+            insert_at_tooltip = random.randint(0, len(base_tooltip))
+            
+            # Generate label and tooltip using list comprehension and join
+            label = ''.join([base_label[:insert_at_label], random.choice(escapes), base_label[insert_at_label:]])
+            tooltip = ''.join([base_tooltip[:insert_at_tooltip], random.choice(escapes), base_tooltip[insert_at_tooltip:]])
+            
+            return {"label": label, "tooltip": tooltip}
+        
+        # Draw the graph using graphviz_draw
+        image = graphviz_draw(
```

Comment:
I think it would be better to test with https://www.rustworkx.org/dev/apiref/rustworkx.PyGraph.to_dot.html

Reply:
sounds good will work on em

Reply:
Do you want me to run that code or ?

Reply:
I want you to avoid writing temporary files to the disk for the unit tests. For the GitHub CI, we found those tests very flaky on macOS. So I suggested that method as a way of testing the fix with only in-memory operations

Reply:
@IvanIsCoding , you want me to adapt my code to use this 
```
with tempfile.TemporaryDirectory() as tmpdirname:
    tmp_path = os.path.join(tmpdirname, 'dag.png')
    dot.write_png(tmp_path)
    image = Image.open(tmp_path)
    os.remove(tmp_path)
```
Or make that entire code as the test case ?

Reply:
Sounds good, Update the code with the recommendation, let me know if it needs any changes